src/cms/views/pages/page_tree_view.py

```diff
@@ -93,15 +88,60 @@ def get(self, request, *args, **kwargs):
                 request, _("You don't have the permission to edit or create pages.")
             )
         context = self.get_context_data(**kwargs)
+
+        pages = region.get_pages(archived=self.archived)
+        enable_drag_and_drop = True
+        # Filter pages according to given filters, if any
+        filter_data = kwargs.get("filter_data")
+        if filter_data is not None:
+            # Set data for filter form rendering
+            filter_form = PageFilterForm(data=filter_data)
+            if filter_form.is_valid():
+                # only filter if at least one checkbox but not all are checked
+                if (
+                    0
+                    < len(filter_form.cleaned_data["translation_status"])
+                    < len(translation_status.CHOICES)
+                ):
+                    enable_drag_and_drop = False
+                    utd = (
+                        translation_status.UP_TO_DATE
+                        in filter_form.cleaned_data["translation_status"]
+                    )
+                    m = (
+                        translation_status.MISSING
+                        in filter_form.cleaned_data["translation_status"]
+                    )
+                    od = (
+                        translation_status.OUTDATED
+                        in filter_form.cleaned_data["translation_status"]
+                    )
+
+                    def f(page):
+                        translation = page.get_translation(language_code)
+                        if translation is None:
+                            return m
+                        return od if translation.is_outdated else utd
```

Comment:
```suggestion
                    def page_filter(page):
                        translation = page.get_translation(language_code)
                        if not translation:
                            return filter_missing
                        elif translation.is_outdated
                            return filter_outdated
                        return filter_up_to_date
```
I find this slightly more readable.